pepper-apis/studybuilder-cli/src/main/java/org/broadinstitute/ddp/studybuilder/task/SimpleActivityRevisionTask.java

```diff
@@ -0,0 +1,335 @@
+    public void run(Handle handle) {
+        this.adminUser = handle.attach(UserDao.class).findUserByGuid(cfg.getString("adminUser.guid")).get();
+
+        this.studyDto = handle.attach(JdbiUmbrellaStudy.class).findByStudyGuid(dataCfg.getString("study.guid"));
+
+        List<? extends Config> activityUpdateConfigs = dataCfg.getConfigList(STUDY_UPDATES);
+
+        this.sqlHelper = handle.attach(SimpleActivityRevisionTask.SqlHelper.class);
+        this.activityDao = handle.attach(ActivityDao.class);
+        this.jdbiVarSubst = handle.attach(JdbiVariableSubstitution.class);
+        this.jdbiRevision = handle.attach(JdbiRevision.class);
+        this.templateDao = handle.attach(TemplateDao.class);
+        this.jdbiBlockContent = handle.attach(JdbiBlockContent.class);
+
+        activitiesUpdate(handle, activityUpdateConfigs);
+    }
+
+    private ActivityVersionDto getVersion(Handle handle, RevisionMetadata meta, String activityCode, String versionTag) {
+        long activityId = ActivityBuilder.findActivityId(handle, studyDto.getId(), activityCode);
+        return activityDao.changeVersion(activityId, versionTag, meta);
+    }
+
+    private void activitiesUpdate(Handle handle, List<? extends Config> activityConfigList) {
+        for (Config activityConfig: activityConfigList) {
+            String activityCode = activityConfig.getString("activityCode");
+            String versionTag = activityConfig.getString("versionTag");
+
+            log.info("Changing version of {} to {} with timestamp={}", activityCode, versionTag, timestamp);
+            long ts = this.timestamp.toEpochMilli();
+
+            String reason = String.format(
+                    "Update activity with studyGuid=%s activityCode=%s to versionTag=%s",
+                    studyDto.getGuid(), activityCode, versionTag);
+            RevisionMetadata metaConsent = new RevisionMetadata(ts, adminUser.getId(), reason);
+
+            ActivityVersionDto activityVersion = getVersion(handle, metaConsent, activityCode, versionTag);
+            activityUpdate(metaConsent, activityVersion, activityConfig);
+        }
+    }
+
+    private void activityUpdate(RevisionMetadata meta, ActivityVersionDto version, Config activityConfig) {
+        updateTemplates(meta, version, activityConfig);
+        updateTemplateVariables(meta, version, activityConfig); //revision only variables in the Template
+        updateQuestionTemplateVariables(meta, version, activityConfig); //revision only variables in the Question Template
+    }
+
+    private void updateTemplates(RevisionMetadata meta, ActivityVersionDto version, Config activityConfig) {
+        log.info("UPDATE Templates/Content Blocks");
+        List<? extends Config> configList = null;
+
+        try {
+            configList = activityConfig.getConfigList(BLOCK_UPDATES);
+        } catch (Exception e) {
+            log.info("No Template updates found for activity={}", version.getActivityId());
+            return;
+        }
```

Comment:
What do you think of a change?
If I initialize `List<? extends Config> configList` early in class (for example, in run() function), and if  `getConfigList` throws an exception, execution would stops early before calling and making any updates. I'm afraid if it throws an exeception  here, something are updated while something else are not.

Reply:
In this case, it is getting a list of configs off of a nested object in the json specific to the activity. `updateActivities` is called with the list that is found from `activity-updates` which then loops through those and passes that config down into each of the three separate methods where they then grab the list of changes off of that config object.

Btw, when a task is run under the StudyBuilderCli, the whole task run is wrapped in a transaction so, with a failure, changes are rolled back. All pass or none pass.

Reply:
Building on what Rich said, this is intending to be a generic update processor and although this exemplar includes a Templates/Content Block,  it shouldn't have to for the other types of updates to run.  Same holds true for each of the other types of updates this supports.

Reply:
I see. I like this a lot. Thank you both for making this happen.